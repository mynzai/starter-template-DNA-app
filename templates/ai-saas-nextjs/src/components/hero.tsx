import Link from 'next/link'
import { Button } from '@/components/ui/button'
import { ArrowRightIcon } from 'lucide-react'

export function Hero() {
  return (
    <section className="relative overflow-hidden gradient-bg py-20 lg:py-32">
      <div className="container mx-auto px-4 sm:px-6 lg:px-8">
        <div className="text-center">
          <h1 className="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl lg:text-7xl dark:text-white">
            Transform Your Business with{' '}
            <span className="bg-gradient-to-r from-blue-600 to-purple-600 bg-clip-text text-transparent">
              AI Power
            </span>
          </h1>
          <p className="mx-auto mt-6 max-w-2xl text-lg leading-8 text-gray-600 dark:text-gray-300">
            Harness the power of artificial intelligence to automate workflows, 
            generate insights, and accelerate your business growth with our comprehensive SaaS platform.
          </p>
          <div className="mt-10 flex items-center justify-center gap-x-6">
            <Button asChild size="lg" className="group">
              <Link href="/auth/signup">
                Start Free Trial
                <ArrowRightIcon className="ml-2 h-4 w-4 transition-transform group-hover:translate-x-1" />
              </Link>
            </Button>
            <Button variant="outline" size="lg" asChild>
              <Link href="/demo">
                Watch Demo
              </Link>
            </Button>
          </div>
          <div className="mt-16 flow-root sm:mt-24">
            <div className="glass-effect -m-2 rounded-xl lg:-m-4 lg:rounded-2xl lg:p-4">
              <div className="aspect-video rounded-md bg-gradient-to-br from-blue-50 to-purple-50 p-8 dark:from-gray-800 dark:to-gray-900">
                <div className="h-full w-full rounded-lg bg-white/50 backdrop-blur-sm dark:bg-gray-800/50">
                  <div className="flex h-full items-center justify-center">
                    <div className="text-center">
                      <div className="mx-auto h-16 w-16 rounded-full bg-gradient-to-r from-blue-500 to-purple-500" />
                      <p className="mt-4 text-sm text-gray-600 dark:text-gray-400">
                        Interactive Demo Coming Soon
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      
      {/* Background decoration */}
      <div className="absolute inset-x-0 -top-40 -z-10 transform-gpu overflow-hidden blur-3xl sm:-top-80">
        <div
          className="relative left-[calc(50%-11rem)] aspect-[1155/678] w-[36.125rem] -translate-x-1/2 rotate-[30deg] bg-gradient-to-tr from-blue-600 to-purple-600 opacity-20 sm:left-[calc(50%-30rem)] sm:w-[72.1875rem]"
          style={{
            clipPath:
              'polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%)',
          }}
        />
      </div>
    </section>
  )
}