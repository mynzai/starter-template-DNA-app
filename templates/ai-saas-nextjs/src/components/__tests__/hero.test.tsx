import { render, screen } from '@testing-library/react'
import { Hero } from '../hero'

// Mock next/link
jest.mock('next/link', () => {
  return ({ children, href, ...props }: any) => {
    return (
      <a href={href} {...props}>
        {children}
      </a>
    )
  }
})

describe('Hero Component', () => {
  it('renders the main heading', () => {
    render(<Hero />)
    
    const heading = screen.getByRole('heading', { level: 1 })
    expect(heading).toBeInTheDocument()
    expect(heading).toHaveTextContent(/Transform Your Business with AI Power/)
  })
  
  it('renders the description text', () => {
    render(<Hero />)
    
    const description = screen.getByText(/Harness the power of artificial intelligence/)
    expect(description).toBeInTheDocument()
  })
  
  it('renders call-to-action buttons', () => {
    render(<Hero />)
    
    const startTrialButton = screen.getByRole('link', { name: /start free trial/i })
    const watchDemoButton = screen.getByRole('link', { name: /watch demo/i })
    
    expect(startTrialButton).toBeInTheDocument()
    expect(watchDemoButton).toBeInTheDocument()
  })
  
  it('has correct links for CTA buttons', () => {
    render(<Hero />)
    
    const startTrialButton = screen.getByRole('link', { name: /start free trial/i })
    const watchDemoButton = screen.getByRole('link', { name: /watch demo/i })
    
    expect(startTrialButton).toHaveAttribute('href', '/auth/signup')
    expect(watchDemoButton).toHaveAttribute('href', '/demo')
  })
  
  it('has proper semantic structure', () => {
    render(<Hero />)
    
    const section = screen.getByRole('banner', { hidden: true }) || 
                   screen.getByText(/Transform Your Business/).closest('section')
    
    expect(section).toBeInTheDocument()
  })
  
  it('renders gradient background elements', () => {
    const { container } = render(<Hero />)
    
    // Check for gradient background classes
    const gradientElements = container.querySelectorAll('.gradient-bg, .bg-gradient-to-r, .bg-gradient-to-br')
    expect(gradientElements.length).toBeGreaterThan(0)
  })
  
  it('is accessible', () => {
    render(<Hero />)
    
    // Check for proper heading hierarchy
    const h1 = screen.getByRole('heading', { level: 1 })
    expect(h1).toBeInTheDocument()
    
    // Check that buttons are properly labeled
    const buttons = screen.getAllByRole('link')
    buttons.forEach(button => {
      expect(button).toHaveAccessibleName()
    })
  })
  
  it('matches snapshot', () => {
    const { container } = render(<Hero />)
    expect(container.firstChild).toMatchSnapshot()
  })
})