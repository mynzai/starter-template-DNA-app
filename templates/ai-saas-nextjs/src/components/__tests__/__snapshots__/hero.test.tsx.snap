// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hero Component matches snapshot 1`] = `
<section
  class="relative overflow-hidden gradient-bg py-20 lg:py-32"
>
  <div
    class="container mx-auto px-4 sm:px-6 lg:px-8"
  >
    <div
      class="text-center"
    >
      <h1
        class="text-4xl font-bold tracking-tight text-gray-900 sm:text-6xl lg:text-7xl dark:text-white"
      >
        Transform Your Business with
         
        <span
          class="bg-gradient-to-r from-blue-600 to-purple-600 bg-clip-text text-transparent"
        >
          AI Power
        </span>
      </h1>
      <p
        class="mx-auto mt-6 max-w-2xl text-lg leading-8 text-gray-600 dark:text-gray-300"
      >
        Harness the power of artificial intelligence to automate workflows, generate insights, and accelerate your business growth with our comprehensive SaaS platform.
      </p>
      <div
        class="mt-10 flex items-center justify-center gap-x-6"
      >
        <a
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow hover:bg-primary/90 h-10 rounded-md px-8 group"
          href="/auth/signup"
        >
          Start Free Trial
          <svg
            class="lucide lucide-arrow-right ml-2 h-4 w-4 transition-transform group-hover:translate-x-1"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5 12h14"
            />
            <path
              d="m12 5 7 7-7 7"
            />
          </svg>
        </a>
        <a
          class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground h-10 rounded-md px-8"
          href="/chat"
        >
          Try AI Chat
        </a>
      </div>
      <div
        class="mt-16 flow-root sm:mt-24"
      >
        <div
          class="glass-effect -m-2 rounded-xl lg:-m-4 lg:rounded-2xl lg:p-4"
        >
          <div
            class="aspect-video rounded-md bg-gradient-to-br from-blue-50 to-purple-50 p-8 dark:from-gray-800 dark:to-gray-900"
          >
            <div
              class="h-full w-full rounded-lg bg-white/50 backdrop-blur-sm dark:bg-gray-800/50"
            >
              <div
                class="flex h-full items-center justify-center"
              >
                <div
                  class="text-center"
                >
                  <div
                    class="mx-auto h-16 w-16 rounded-full bg-gradient-to-r from-blue-500 to-purple-500"
                  />
                  <p
                    class="mt-4 text-sm text-gray-600 dark:text-gray-400"
                  >
                    Interactive Demo Coming Soon
                  </p>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="absolute inset-x-0 -top-40 -z-10 transform-gpu overflow-hidden blur-3xl sm:-top-80"
  >
    <div
      class="relative left-[calc(50%-11rem)] aspect-[1155/678] w-[36.125rem] -translate-x-1/2 rotate-[30deg] bg-gradient-to-tr from-blue-600 to-purple-600 opacity-20 sm:left-[calc(50%-30rem)] sm:w-[72.1875rem]"
      style="clip-path: polygon(74.1% 44.1%, 100% 61.6%, 97.5% 26.9%, 85.5% 0.1%, 80.7% 2%, 72.5% 32.5%, 60.2% 62.4%, 52.4% 68.1%, 47.5% 58.3%, 45.2% 34.5%, 27.5% 76.7%, 0.1% 64.9%, 17.9% 100%, 27.6% 76.8%, 76.1% 97.7%, 74.1% 44.1%);"
    />
  </div>
</section>
`;
